distributed/worker.py

```diff
@@ -3190,16 +3198,18 @@ def apply_function_actor(
     with active_threads_lock:
         active_threads[ident] = key
 
-    thread_state.execution_state = execution_state
-    thread_state.key = key
-    thread_state.actor = True
-
-    result = function(*args, **kwargs)
```

Comment:
State was actually leaking all over the place making get_worker in our test suite extremely unreliable.